tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/RetryConditionTest.java

```diff
@@ -215,4 +227,144 @@ public void testClassLevelRetryWithAbortOnTrue() {
         Assert.assertEquals(clientForClassLevelAbortOn.getRetryCountForWritingService(), 1,
             "The max invocation counter should be 1 as the abort condition is true");
     }
+
+
+    /**
+     * Persistent Error condition. Will retry 2 times and still throw exception.
+     * ServiceA uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and will always return IOException.
+     */
+    @Test
+    public void testAsyncRetryExceptionally() {
+        final CompletionStage<String> future = asyncRetryClient.serviceA();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServA());
+    }
+
+    /**
+     * Persistent Error condition inside a CompletableFuture.
+     * Will not retry because method is not marked with @Asynchronous
+     * ServiceB will always complete exceptionally with IOException.
+     */
+    @Test
+    public void testNoAsynWilNotRetryExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceBFailExceptionally(future);
```

Comment:
What's the reason for passing in `future` here, rather than creating it inside `serviceBFailExceptionally`?

Reply:
It's just an additional scenario where an uncompleted future is passes as a parameter. Probably it doesn't add much.